melange-opam-template.opam

```diff
@@ -7,14 +7,25 @@ license: "MIT"
 homepage: "https://github.com/your/project"
 bug-reports: "https://github.com/your/project/issues"
 depends: [
-  "ocaml" {>= "4.14.0" & < "5.0.0"}
+  "ocaml" {>= "5.1.0"}
   "reason" {>= "3.8.1" & < "4.0.0"}
   "dune" {>= "3.8"}
   "melange" {>= "1.0.0"}
-  "reactjs-jsx-ppx" {>= "1.0.0"}
+  "mlx" {= "~dev"}
+  "ocamlmerlin-mlx" {= "~dev"}
+  "reason-react-ppx" {= "~dev"}
   "reason-react" {>= "0.11.0"}
   "ocaml-lsp-server" {dev}
   "dot-merlin-reader" {dev}
   "odoc" {with-doc}
 ]
+pin-depends: [
+  ["dune.3.10.0"                  "git+https://github.com/andreypopp/dune.git#main"]
+  ["melange.1.0.0"                "git+https://github.com/melange-re/melange.git#main"]
+  ["mlx.~dev"                     "https://github.com/andreypopp/mlx/archive/refs/heads/main.zip"]
+  ["ocamlmerlin-mlx.~dev"         "https://github.com/andreypopp/mlx/archive/refs/heads/main.zip"]
+  ["reason.3.8.2"                 "git+https://github.com/reasonml/reason.git#master"]
```

Comment:
Reason still needed to build reason-react right? but using mlx with melange would not require it?

Reply:
Yeah, needed for reason-react.

I'm actually wondering if some core libs in Melange ecosystem should be in Reason or pure OCaml. Seems like bindings to DOM and other Web APIs don't require JSX and thus can be in .ml (just less things to build to use them).

Reply:
All Melange "core" libs like `melange.dom` and `melange.node` have always been [written in `ml`](https://github.com/melange-re/melange/tree/b334d3a4d8fbda1b3c4e10939525cc58268544dd/jscomp/others). And some libs like [melange-jest](https://github.com/melange-community/melange-jest/blob/main/jest/jest.ml) are also in ml. But as soon as there was some JSX involved, it was natural to switch to Reason.

I agree that now, with the appearance of `mlx`, there is a much stronger case for writing things like ReactJS bindings in OCaml syntax.